src/Servers/Kestrel/Core/src/Internal/Http3/Http3FrameWriter.cs

```diff
@@ -281,6 +327,20 @@ internal void WriteResponseHeaders(int statusCode, IHeaderDictionary headers)
             }
         }
 
+        private int CalculateHeaderSize(IEnumerable<KeyValuePair<string, string>> enumerable)
+        {
+            var size = 0;
+            // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-header-size-constraints
+            // The size of a header field list is calculated based on the uncompressed size of header fields,
+            // including the length of the name and value in bytes plus an overhead of 32 bytes for each header field.
+            foreach (var header in enumerable)
```

Comment:
I'd be ideal if we could calculate this during the encode process to avoid enumerating twice. E.g. if the QPackEncoder kept a running total for us. Then we could check the total and abort before flushing the headers.

Reply:
Sure, but I'd prefer to do this in a follow up issue. It's a bit much for this PR as well as needing to react in the runtime repo.